salalib/axialmap.cpp

```diff
@@ -2135,27 +2140,31 @@ bool ShapeGraph::angularstepdepth(Communicator *comm)
       }
    }
    int depthlevel = 0;
+   auto binIter = bins.begin();
    int currentbin = 0;
    while (opencount) {
-      while (!bins[currentbin].size()) {
+      while (binIter->empty()) {
          depthlevel++;
+         binIter++;
          currentbin++;
-         if (currentbin == tulip_bins) {
-            currentbin = 0;
+         if (binIter == bins.end()) {
+            binIter = bins.begin();
          }
       }
       SegmentData lineindex;
-      if (bins[currentbin].size() > 1) {
+      if (binIter->size() > 1) {
          // it is slightly slower to delete from an arbitrary place in the bin,
          // but it is necessary to use random paths to even out the number of times through equal paths
-         int curr = pafrand() % bins[currentbin].size();
-         lineindex = bins[currentbin][curr];
-         bins[currentbin].remove_at(curr);
+         int curr = pafrand() % binIter->size();
+         auto currIter = binIter->begin() + curr;
+         lineindex = *currIter;
+         binIter->erase(currIter);
          // note: do not clear choice values here!
       }
       else {
-         lineindex = bins[currentbin][0];
-         bins[currentbin].pop_back();
+         // PK: Not sure why pick front element but erase last...

```

Comment:
This doesn't matter - the size is 1, so `front` and `back` are the same element. Probably `pop_front` did not exist when this code was written originally 🙄